packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2511,108 +2567,70 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
     private submit(
         type: ContainerMessageType,
-        content: any,
+        contents: any,
         localOpMetadata: unknown = undefined,
-        opMetadata: Record<string, unknown> | undefined = undefined,
+        metadata: Record<string, unknown> | undefined = undefined,
     ): void {
         this.verifyNotClosed();
 
         // There should be no ops in detached container state!
         assert(this.attachState !== AttachState.Detached, 0x132 /* "sending ops in detached container" */);
 
-        let clientSequenceNumber: number = -1;
-        let opMetadataInternal = opMetadata;
-
-        if (this.canSendOps()) {
-            const serializedContent = JSON.stringify(content);
-
-            // If in TurnBased flush mode we will trigger a flush at the next turn break
-            if (this.flushMode === FlushMode.TurnBased && !this.needsFlush) {
-                opMetadataInternal = {
-                    ...opMetadata,
-                    batch: true,
-                };
-                this.needsFlush = true;
-
+        const deserializedContent: ContainerRuntimeMessage = { type, contents };
+        const serializedContent = JSON.stringify(deserializedContent);
+
+        if (serializedContent.length > defaultMaxOpSizeInBytes) {
+            // If the content length is larger than the client configured message size
+            // instead of splitting the content, we will fail by explicitly closing the container
+            const error = new GenericError(
+                "OpTooLarge",
+                /* error */ undefined,
+                {
+                    length: serializedContent.length,
+                    limit: defaultMaxOpSizeInBytes,
+                });
+            this.closeFn(error);
+            throw error;
+        } else {
```

Comment:
#nit. The `then` branch will throw anyway, so there's no need for the `else` statement and this would decrease branching/identation bbelow.